src/ext/respToRedisLoader.c

```diff
@@ -195,43 +194,80 @@ static void redisLoaderDelete(void *context) {
     RDB_free(ctx->p, ctx);
 }
 
+int redisAuth(RdbxRespToRedisLoader *ctx, RdbxRedisAuth *auth) {
+    int numTokens = 0, len = 0;
+    char buf[400];
+    if ((auth->pwd == NULL) && (auth->authCmd == NULL))
+        return 0;
+
+    if (auth->pwd) { /* AUTH [username] password */
+        if (auth->user) {
+            len = snprintf(buf, sizeof(buf) - 1, "*3\r\n$4\r\nauth\r\n$%ld\r\n%s\r\n$%ld\r\n%s\r\n",
+                           strlen(auth->user), auth->user, strlen(auth->pwd), auth->pwd);
+        } else {
+            len = snprintf(buf, sizeof(buf) - 1, "*2\r\n$4\r\nauth\r\n$%ld\r\n%s\r\n", strlen(auth->pwd), auth->pwd);
```

Comment:
just out of curiosity, why do you pass `sizeof(buf) - 1` ? IIUC, `-1` is not going to help with anything. `snprintf` will write null terminator all the time, no?

Reply:
Probably [Win legacy issue](https://stackoverflow.com/questions/7706936/is-snprintf-always-null-terminating) that I carry :)